### PR TITLE
Form integration compatable

### PR DESCRIPTION
**Pull request by meinfr13nd**
Makes API responses compatible with the current API spec proposal
https://docs.google.com/document/d/1ypoLhyBFQx_O77DBh7Rj-gX9gO2b4tPxkTuNVJBseFs/edit#heading=h.s267qjeh2sp7

### CODE REVIEW

File: apps/javaVendor/src/TextInputComponent.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+
+public class TextInputComponent {
```
Comment (GrantMatejka) [inline apps/javaVendor/src/TextInputComponent.java]: this component seems like a great opportunity for some design patterns practice, we're duplicating a lot of code

**Comment by GrantMatejka** [comment]
Made a request for some changes but overall looks great and I'm glad we can get this java app evolving

**Comment by GrantMatejka** [comment]
Also thanks for updating the readme 🌽

File: apps/javaVendor/src/TextInputComponent.java (hunk 1 above)
Comment (meinfr13nd) [reply apps/javaVendor/src/TextInputComponent.java]: Yeah, I think I'll take a look at the factory pattern and ping you when its done

Comment (meinfr13nd) [reply apps/javaVendor/src/TextInputComponent.java]: Added the factory pattern to clean things up

**Comment by GrantMatejka** [comment]
Awesome, looks much better 🌽

**Review by GrantMatejka — APPROVED** [review]

**Pull request merged**